src/OrleansCodeGenerator/CodeGeneratorCommon.cs

```diff
@@ -87,9 +89,13 @@ public static byte[] CompileAssembly(GeneratedSyntax generatedSyntax, string ass
                     .AddSyntaxTrees(code.SyntaxTree)
                     .AddReferences(assemblies)
                     .WithOptions(options);
+            MemoryStream symbolStream = null;
             using (var stream = new MemoryStream())
+#if DEBUG
+            using (symbolStream = new MemoryStream())
+#endif
```

Comment:
What I meant (not sure if it's possible, or even easy, otherwise it might not be worth it), is to generate debug symbols if you are codegening types for an assembly that (or a set of assemblies where at least one) is debuggable.
Also, we shouldn't change the behavior that prints whether Orleans itself is debuggable. Now this will print Debug if the entry assembly is debuggable, instead of whether Orleans itself is.